pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java

```diff
@@ -212,12 +217,27 @@ private boolean evalStudyQuery(InterpreterContext ictx, StudyQueryContext ctx) {
         return applyStudyPredicate(ictx, ctx.studyPredicate(), umbrellaStudyGuid);
     }
 
-    private boolean applyStudyPredicate(InterpreterContext ictx, StudyPredicateContext predCtx, String umbrellaStudyGuid) {
-        Optional<GovernancePolicy> policy = ictx.getHandle().attach(StudyGovernanceDao.class).findPolicyByStudyGuid(umbrellaStudyGuid);
-        String msg = "Governance policy for " + umbrellaStudyGuid + " doesn't exist";
-        return policy.map(
-                p -> p.hasReachedAgeOfMajority(ictx.getHandle(), new TreeWalkInterpreter(), ictx.getUserGuid())
-        ).orElseThrow(() -> new PexFetchException(new NoSuchElementException(msg)));
+    private boolean applyStudyPredicate(InterpreterContext ictx, StudyPredicateContext predCtx, String studyGuid) {
+        if (predCtx instanceof PexParser.HasAgedUpPredicateContext) {
+            GovernancePolicy policy = ictx.getHandle().attach(StudyGovernanceDao.class)
+                    .findPolicyByStudyGuid(studyGuid)
+                    .orElse(null);
+            if (policy == null) {
+                throw new PexFetchException(new NoSuchElementException("Governance policy for " + studyGuid + " does not exist"));
+            }
+
+            String userGuid = ictx.getUserGuid();
+            UserProfileDto profileDto = ictx.getHandle().attach(JdbiProfile.class).getUserProfileByUserGuid(userGuid);
+            if (profileDto == null || profileDto.getBirthDate() == null) {
+                LOG.warn("User {} in study {} does not have profile or birth date to evaluate age-up policy, defaulting to false",
+                        userGuid, studyGuid);
```

Comment:
There's an overridden version of `hasReachedAgeOfMajority()` that is a convenience method, which is the one we were using before and which will thrown exception. I changed it to use the other version which doesn't throw, so we have better control since there might be cases where we might not have birthdate yet.

In short, this PR should have the right thing -- the one that doesn't throw exception.